src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandBinder.cs

```diff
@@ -821,15 +824,7 @@ private Expression CreateTotalCountExpression(Expression source, bool? countOpti
             {
                 if (!structuralProperty.Type.IsCollection())
                 {
-                    // EF5 and EF6 doesn't support to compare the complex with null.
-                    // With the following null check, EF5 and EF6 will throw except similar to:
-                    // "Cannot compare elements of type 'xxxx'.Only primitive types, enumeration types and entity types are supported.
-                    // EF Core has imporvement so the following null won't throw error.
-                    if (_dataSourceProviderKind == DataSourceProviderKind.EFCore ||
```

Comment:
This check is unnecessary - if the condition is true we could already guarantee that `nullCheck` was `null` and so the assignment is essentially a no-op